### ISSUE TITLE
NSP reports security vulnerability in hoek

### ISSUE THREAD

**Issue by akselnordal** [vvo/selenium-standalone#360]
## Expected Behavior
NSP should not report any security vulnerabilities

## Current Behavior
When installing the latest `selenium-standalone` package from npm, then running `nsp check`, it gives the following output:

```
(+) 2 vulnerabilities found
┌────────────┬────────────────────────────────────────────────────────────────────┐
│            │ Prototype pollution attack                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Name       │ hoek                                                               │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ CVSS       │ 4 (Medium)                                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Installed  │ 2.16.3                                                             │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Vulnerable │ <= 4.2.0 || >= 5.0.0 < 5.0.3                                       │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Patched    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                        │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Path       │ XXX > nsp@3.2.1 > wreck@12.5.1 > boom@5.2.0 > hoek@2.16.3      │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ More Info  │ https://nodesecurity.io/advisories/566                             │
└────────────┴────────────────────────────────────────────────────────────────────┘

┌────────────┬────────────────────────────────────────────────────────────────────┐
│            │ Prototype pollution attack                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Name       │ hoek                                                               │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ CVSS       │ 4 (Medium)                                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Installed  │ 2.16.3                                                             │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Vulnerable │ <= 4.2.0 || >= 5.0.0 < 5.0.3                                       │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Patched    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                        │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Path       │ XXX > selenium-standalone@6.14.0 > request@2.79.0 > hawk@3.1.3 │
│            │ > hoek@2.16.3                                                      │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ More Info  │ https://nodesecurity.io/advisories/566                             │
└────────────┴────────────────────────────────────────────────────────────────────┘
```

## Steps to Reproduce
Provide an unambiguous set of steps to reproduce this bug. Include commands that you used
in the shell, or snippets of how you are using the JavaScript API.

1.  npm init
2.  npm install selenium-standalone --save
3.  npm install nsp
4.  ./node_modules/.bin/nsp check

NB: nsp will by default only check for production `dependencies`, not `devDependencies`. And since selenium is usually ran as a devDepenceny, you might say that it isn't important to fix. But since devDependencies also may inject security flaws into production code, it would be nice to have this dependency fixed.

To run nsp with checking devDepencies, this small script may be added to the root

And in the root folder, add this file: `nsp-preprocessor-dev-depencies.js`

```
const pkg = require('./package.json');

function check(args) {
  // do something to read or generate package.json, npm-shrinkwrap.json and package-lock.json
  // the path to the project can be found as `args.path`
  // `pkg` must be the JSON parsed contents of package.json
  // `shrinkwrap` must be the JSON parsed contents of npm-shrinkwrap.json, if it exists. this may be left out.
  // `packagelock` must be the JSON parsed contents of package-lock.json, if it exists. this may also be left out.
  // return Object.assign(args, { pkg, shrinkwrap, packagelock });
  const allDeps = Object.assign({}, pkg, {
    dependencies: pkg.devDependencies,
  });
  const argsWithAllDeps = Object.assign(args, { pkg: allDeps });
  return argsWithAllDeps;
}

module.exports = {
  check,
};
```

And then run the nsp-check like this
```
./node_modules/.bin/nsp check --preprocessor dev-dependencies
```

## Your Environment
* Version of `selenium-standalone` that you are using:
`selenium-standalone": "^6.14.0`
* Is there another tool calling `selenium-standalone` on your behalf:
no (but here are several tools using `selenium-standalone`, such as wdio-selenium-standalone-service - https://github.com/webdriverio/wdio-selenium-standalone-service/issues/49)
* System/platform:
OSx, Node@8.10.0
* Other details that might be important:

**Comment by warpech** [comment]
This is fixed in PR #359. 

@vvo pls consider merging ASAP

**Comment by serbanghita** [comment]
I'll close this since there are no vulns reported. @akselnordal thanks for reporting this!

PS: Funny enough if you run the nsp check on the devDependencies now it reports that nsp has a "vulnerability" - lodash prototype pollution :)

```
┌────────────┬────────────────────────────────────────────────────────────────────┐
│            │ Prototype Pollution                                                │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Name       │ lodash                                                             │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ CVSS       │ 2 (Low)                                                            │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Installed  │ 3.10.1                                                             │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Vulnerable │ <4.17.5                                                            │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Patched    │ >=4.17.5                                                           │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Path       │ selenium-standalone@6.15.1 > nsp@3.2.1 > cli-table2@0.2.0 >        │
│            │ lodash@3.10.1                                                      │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ More Info  │ https://nodesecurity.io/advisories/577                             │
└────────────┴────────────────────────────────────────────────────────────────────┘
```
